src/vip/batch_address_test_tool/queue.clj

```diff
@@ -25,18 +51,29 @@
           (do (Thread/sleep (* attempt 1000))
               (recur (inc attempt)))
           (do (log/error "Connecting to RabbitMQ failed. Bailing.")
-            (throw (ex-info "Connecting to RabbitMQ failed" {:attemts attempt})))))))
+            (throw (ex-info "Connecting to RabbitMQ failed" {:attempts attempt})))))))
   (reset! rabbit-channel
           (let [ch (lch/open @rabbit-connection)]
             (le/topic ch (config [:rabbitmq :exchange]) {:durable false :auto-delete true})
             (log/info "RabbitMQ topic set.")
+            (setup-consumer ch handler-fn)
             ch)))
 
 (defn publish
```

Comment:
`publish` and `publish-event` would probably confuse me  - the `payload` arg is either first or second depending on whether it's a queue or routing key to which we're publishing.